docs/current_docs/manuals/developer/cookbook.mdx

```diff
@@ -668,3 +746,37 @@ Set environment variables in a container:
 ```shell
 dagger call set-env
 ```
+
+### Persist service state across runs
+
+The following Dagger Function uses a cache volume to persist a Redis service's data across Dagger Function runs.
+
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./snippets/services/persist-service-state/go/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./snippets/services/persist-service-state/python/main.py
+```
+
+</TabItem>
+<TabItem value="TypeScript">
+
+```typescript file=./snippets/services/persist-service-state/typescript/index.ts
+```
+
+</TabItem>
+</Tabs>
+
+#### Example
+
+Create a Redis service with a cache volume to persist its data:
+
+```shell
+dagger call redis-service
```

Comment:
@vito @kpenfound I'm not sure if I'm testing this wrong. I ran it once, then commented the `set/save` commands in the function and ran it again, expecting to see the output `abc` since the value should be stored in the cache from the first run. However no output was returned on the second run. Am I missing something or is this code broken?

Reply:
@vikram-dagger seems like the issue is coming from here: https://github.com/dagger/dagger/blob/main/docs/current_docs/manuals/developer/snippets/services/persist-service-state/go/main.go#L15. If you remove the `WithWorkdir` step, that works

Reply:
@marcosnils I tried it after removing that line but it still doesn't work when I comment out the `save` on the second run. The value `abc` doesn't appear in the output.

Reply:
This line probably doesn't need to block this PR anyway since the related snippet isn't part of this PR unless I'm misunderstanding @vikram-dagger @marcosnils

Reply:
Reviewing the snippet that Marcos linked, it looks like that isn't intended to show caching with consecutive executions, it shows it all in one run. The `return redisCLI. {get foo}` shows the cache volume since redisCLI wasn't reassigned to itself in the `set` above it